react/style-guide.md

```diff
@@ -0,0 +1,948 @@
+# Javascript
+
+We use the [Airbnb Javascript Style Guide](https://github.com/airbnb/javascript) as reference for Javascript style.
+
+# React
+
+For all React technologies, we have our own styleguide also heavily influenced by [Airbnb React Style Guide](https://github.com/airbnb/javascript/tree/master/react).
+
+## Table of Contents
+
+  1. [Basic Rules](#basic-rules)
+  1. [Folder Structure](#folder-structure)
+  1. [Class vs `React.createClass` vs stateless](#class-vs-reactcreateclass-vs-stateless)
+  1. [Mixins](#mixins)
+  1. [Naming](#naming)
+  1. [Import](#import)
+  1. [Declaration](#declaration)
+  1. [Alignment](#alignment)
+  1. [Quotes](#quotes)
+  1. [Spacing](#spacing)
+  1. [Props](#props)
+  1. [Refs](#refs)
+  1. [Parentheses](#parentheses)
+  1. [Tags](#tags)
+  1. [Methods](#methods)
+  1. [Ordering](#ordering)
+  1. [`isMounted`](#ismounted)
+
+## Basic Rules
+
+  - Only include one React component per file.
+  - Always use JSX syntax.
+  - Do not use `React.createElement` unless you're initializing the app from a file that is not JSX.
+  - Always use export default for Components.
+
+## Folder Structure
+
+```
+src  
+│
+└───app
+│   │
+│   └───components
+│   | └───BaseComponents
+│   | | └───Input
+│   | | └───Text
+│   | | └───Button
+│   | | └───etc
+│   └───screens
+| | └───aScreenComponent
+| | | └───components
+| | | | constants.js
+| | | | i18n.js
+| | | | index.js
+| | | | layout.js
+| | | | styles.scss
+| | | | utils.js
+└───assets
+└───config
+└───constants
+└───redux
+│   └───aReducer
+│   │   | actions.js
+│   │   | reducer.js
+│   
+└───scss
+└───services
+└───utils
+│   index.js
+```
+
+## Class vs `React.createClass` vs stateless
+
+  - If you have internal state and/or refs, prefer `class extends React.Component` over `React.createClass`. eslint: [`react/prefer-es6-class`](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prefer-es6-class.md) [`react/prefer-stateless-function`](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prefer-stateless-function.md)
+
+    ```jsx
+    // bad
+    const Listing = React.createClass({
+      // ...
+      render() {
+        return <div>{this.state.hello}</div>;
+      }
+    });
+
+    // good
+    class Listing extends React.Component {
+      // ...
+      render() {
+        return <div>{this.state.hello}</div>;
+      }
+    }
+    ```
+
+    And if you don't have state or refs, prefer normal functions (not arrow functions) over classes:
+
+    ```jsx
+    // bad
+    class Listing extends React.Component {
+      render() {
+        return <div>{this.props.hello}</div>;
+      }
+    }
+
+    // bad (relying on function name inference is discouraged)
```

Comment:
I prefer arrow function declaration instead of normal function declaration, it's simpler, cleaner and keeps the "functional programming" that React is trying to encourage

Reply:
I think arrow functions don't lack expressiveness when it comes to declaration. It may be a matter of preference, but with arrow functions we can follow the ES conventions and encourage the functional programming to stay in the same React scope